source/aws-bootstrap-kit/lib/account.ts

```diff
@@ -84,77 +84,132 @@ export class Account extends Construct {
+      onCreate: tagAccount,
+      onUpdate: tagAccount,
+      onDelete: {
+        service: "Organizations",
+        action: "untagResource",
+        region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
+        parameters: {
+          ResourceId: accountId,
+          TagKeys: tags.map(t => t.Key)
+        }
+      },
+      policy: cr.AwsCustomResourcePolicy.fromSdkCalls({
+        resources: cr.AwsCustomResourcePolicy.ANY_RESOURCE,
+      })
+    });
 
-    accountProps.id = accountId;
     this.accountName = accountProps.name;
     this.accountId = accountId;
     this.accountStageName = accountProps.stageName;
+    this.accountStageOrder = accountProps.stageOrder;
 
-    new ssm.StringParameter(this, `${accountProps.name}-AccountDetails`, {
+    let ssmParam = new ssm.StringParameter(this, exitingAccount?`${accountId}-AccountDetails`:`${accountProps.name}-AccountDetails`, {
       description: `Details of ${accountProps.name}`,
       parameterName: `/accounts/${accountProps.name}`,
+      simpleName: false,
       stringValue: JSON.stringify(accountProps),
     });
+    ssmParam.node.addDependency(account);
 
     if (accountProps.parentOrganizationalUnitId) {
-      let parent = new cr.AwsCustomResource(this, "ListParentsCustomResource", {
-        onCreate: {
-          service: "Organizations",
-          action: "listParents",
-          physicalResourceId: cr.PhysicalResourceId.fromResponse(
-            "Parents.0.Id"
-          ),
-          region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
-          parameters: {
-            ChildId: accountId,
-          },
-        },
-        onUpdate: {
-          service: "Organizations",
-          action: "listParents",
-          physicalResourceId: cr.PhysicalResourceId.fromResponse(
-            "Parents.0.Id"
-          ),
-          region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
-          parameters: {
-            ChildId: accountId,
-          },
-        },
-        onDelete: {
-          service: "Organizations",
-          action: "listParents",
-          physicalResourceId: cr.PhysicalResourceId.fromResponse(
-            "Parents.0.Id"
-          ),
-          region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
-          parameters: {
-            ChildId: accountId,
-          },
+      const listParents = {
+        service: "Organizations",
+        action: "listParents",
+        physicalResourceId: cr.PhysicalResourceId.fromResponse(
+          "Parents.0.Id"
+        ),
+        region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
+        parameters: {
+          ChildId: accountId,
         },
+      };
+      let parent = new cr.AwsCustomResource(this, "ListParentsCustomResource", {
+        onCreate: listParents,
+        onUpdate: listParents,
```

Comment:
removing onDelete and onUpdate, as we only move accounts during creation.
Should we be able to move an account onUpdate ?

Reply:
If someone wants to reorganize the hierarchy, he must be able to move the account to another OU or the root:

- Move accounts, remove parent OU
- Deploy
- Create parent OU, move accounts
- Deploy

Since we maintain node.addDependency it should work out of the box.

I haven't found anything besides moveAccount and tagResource in the organization API. (There's also another account management API that has some other features and must be enabled on the organization....)

Reply:
Yes, moving accounts works.